### DISCUSSION TITLE
Max Image Width/Height settings

### DISCUSSION THREAD

**Discussion by angrybrad** [craftcms/cms#8115] [Ideas]
> Created by: Brandon Kelly (brandon@pixelandtonic.com) on 2015/03/06 03:54:45 +0000
> Votes at time of UserVoice import: 40

-----------------------------------

It would be nice if it was possible to limit the width/height of images uploaded to Craft. That way if a really big image is uploaded, beyond any size you'd ever reasonably expect to serve on the website, Craft could automatically resize it down before saving, so they're not wasting so much space on the disk.

**Comment by angrybrad** [comment]
> Posted by Thijs Kaspers (thijs@24creative.nl) on 2015/03/11 08:43:15 +0000
-----------------------------------

Update: Plugin is almost finished for production. Having some issues with variable subfolders, e.g. {slug} set as a subfolder at the Asset Field settings, but with some help of Andris I&#39;ll figure it out these days :-)

**Comment by angrybrad** [comment]
> Posted by Thijs Kaspers (thijs@24creative.nl) on 2015/03/11 08:43:15 +0000
-----------------------------------

I’m working on a plugin to do just this, should be released soon. Of course it will be cool if this is a native Craft feature, but untill then…

**Comment by piotrpog** [comment]
If anyone is interested, this plugin allows for resizing images upon upload:

https://plugins.craftcms.com/image-resizer
